### PR TITLE
[DOCS] Note assumptions for shard size and count recommendations

### LINKED ISSUE #76328: Shard count recommendation does not apply to all use cases
https://www.elastic.co/guide/en/elasticsearch/reference/current/size-your-shards.html#shard-count-recommendation

The above recommendation treats all shards as equal.  For use cases where there can be a lot of small/empty shards like Enterprise Search (depending the specific features being used), indices that are rarely used and are almost empty do not contribute to the heap usage much.  

I will suggest updating the above doc section to clarify that the recommendation assumes that shard sizes are 10-50G each as suggested in the [shard sizing recommendation](https://www.elastic.co/guide/en/elasticsearch/reference/current/size-your-shards.html#shard-size-recommendation) in the same guide.

### PR DESCRIPTION
**Pull request by jrodewig**
On the "Size your shards" page, the shard size recommendation assumes a time
series use case. Similarly, users shouldn't count nearly empty and rarely used
Enterprise Search system indices against the recommended shard count limit.

Closes #76328.

### Preview
https://elasticsearch_76353.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/size-your-shards.html#shard-size-recommendation

### CODE REVIEW

**Review by henningandersen — COMMENTED** [review]
I think the first part of the PR is fine, but worry about the shard count changes. We should probably move that discussion to the issue.

File: docs/reference/how-to/size-your-shards.asciidoc [hunk 1]
```diff
@@ -159,11 +161,13 @@ index                                 prirep shard store
...
 ==== Aim for 20 shards or fewer per GB of heap memory
 
-The number of shards a node can hold is proportional to the node's
-heap memory. For example, a node with 30GB of heap memory should
-have at most 600 shards. The further below this limit you can keep your nodes,
-the better. If you find your nodes exceeding more than 20 shards per GB,
-consider adding another node.
+The number of shards a node can hold is proportional to the node's heap memory.
+However, {es} doesn't enforce a fixed limit. If your shards are consistently
+between 10–50GB, we recommend limiting nodes to 20 shards or fewer per GB of
```
[the remark this answers is not in the document]
Comment (jrodewig) [reply docs/reference/how-to/size-your-shards.asciidoc:166]: Thanks for the feedback @henningandersen. We can discuss this more in #76328 if wanted.

**Comment by jrodewig** [comment]
Moving this PR back to draft while we discuss more in #76328.

File: docs/reference/how-to/size-your-shards.asciidoc [hunk 2]
```diff
@@ -128,8 +128,10 @@ Large shards may make a cluster less likely to recover from failure. When a node
 fails, {es} rebalances the node's shards across the data tier's remaining nodes.
 Large shards can be harder to move across a network and may tax node resources.
 
-While not a hard limit, shards between 10GB and 50GB tend to work well. You may
-be able to use larger shards depending on your network and use case.
+While not a hard limit, shards between 10GB and 50GB tend to work well for logs
+and similar time series data. You may be able to use larger shards depending on
```
Comment (henningandersen) [inline docs/reference/how-to/size-your-shards.asciidoc:132]:
```suggestion
and time series data. You may be able to use larger shards depending on
```

**Review by henningandersen — APPROVED** [review]
LGTM.

**Comment by jrodewig** [comment]
Thanks @henningandersen.

**Pull request merged**